curator-recipes/src/main/java/org/apache/curator/framework/recipes/locks/InterProcessLock.java

```diff
@@ -44,7 +44,7 @@
     /**
      * Perform one release of the mutex.
      *
-     * @throws Exception ZK errors, interruptions, current thread does not own the lock
+     * @throws Exception ZK errors, interruptions
```

Comment:
On the one hand it seems true that not all implementations throw exception in case of different thread, on the other hand, after removing this comment from the interface - its users cannot be aware of the possibility that this exception can be thrown (depending on the implementation). So user that did not create this InterProcessLock, but calls release() from different thread, might be surprised of the IllegalMonitorStateException (in case it is InterProcessMutex, for instance).
Maybe add above the interface InterProcessLock top level javadoc (empty currently):
NOTE: depending on its implementation, InterProcessLock#release() might throw exception if current thread does not own the lock

Reply:
That makes sense, updating the PR

Reply:
updated